lightning/src/ln/channelmanager.rs

```diff
@@ -5864,7 +5888,21 @@ where
 												match next_hop {
 													onion_utils::Hop::Receive(hop_data) => {
 														let current_height: u32 = self.best_block.read().unwrap().height;
-														match create_recv_pending_htlc_info(hop_data,
+														match create_recv_pending_htlc_info(msgs::InboundOnionPayload::Receive(hop_data),
+															incoming_shared_secret, payment_hash, outgoing_amt_msat,
+															outgoing_cltv_value, Some(phantom_shared_secret), false, None,
+															current_height)
+														{
+															Ok(info) => phantom_receives.push((
+																prev_short_channel_id, prev_counterparty_node_id, prev_funding_outpoint,
+																prev_channel_id, prev_user_channel_id, vec![(info, prev_htlc_id)]
+															)),
+															Err(InboundHTLCErr { err_code, err_data, msg }) => failed_payment!(msg, err_code, err_data, Some(phantom_shared_secret))
+														}
+													},
+													onion_utils::Hop::BlindedReceive(hop_data) => {
```

Comment:
Looks like you can extract `hop_data` like
```rust
let hop_data = match next_hop {
	onion_utils::Hop::Receive(hop_data) => msgs::InboundOnionPayload::Receive(hop_data),
	onion_utils::Hop::BlindedReceive(hop_data) => msgs::InboundOnionPayload::BlindedReceive(hop_data),
	_ => panic!()
};
```
and avoid repeating the code here. Maybe other areas where this commit could be DRY'd as well.